SimCalorimetry/HGCalSimProducers/src/HGCHEbackDigitizer.cc

```diff
@@ -11,27 +11,300 @@
 using namespace hgc_digi;
 using namespace hgc_digi_utils;
 
-//
+
+void HGCHEbackSignalScaler::setDoseMap(const std::string& fullpath)
+{
+  doseMap_ = readDosePars(fullpath);
+}
+
+void HGCHEbackSignalScaler::setGeometry(const CaloSubdetectorGeometry* geom)
+{
+  hgcalGeom_ = static_cast<const HGCalGeometry*>(geom);
+}
+
+std::map<int, HGCHEbackSignalScaler::DoseParameters> HGCHEbackSignalScaler::readDosePars(const std::string& fullpath)
+{
+  std::map<int, DoseParameters> result;
+
+  //no dose file means no aging
+  if(fullpath.empty())
+    return result;
+
+  edm::FileInPath fp(fullpath);
+  std::ifstream infile(fp.fullPath());
+  if(!infile.is_open())
+  {
+    throw cms::Exception("FileNotFound") << "Unable to open '" << fullpath << "'" << std::endl;
+  }
+  std::string line;
+  while(getline(infile,line))
+  {
+    int layer;
+    DoseParameters dosePars;
+
+    //space-separated
+    std::stringstream linestream(line);
+    linestream >> layer >> dosePars.a_ >>  dosePars.b_ >> dosePars.c_ >> dosePars.d_ >> dosePars.e_ >> dosePars.f_;
+
+    result[layer] = dosePars;
+  }
+  return result;
+}
+
+double HGCHEbackSignalScaler::getDoseValue(int layer, float radius)
+{
+  double cellDose = std::pow(10, doseMap_[layer].a_ + doseMap_[layer].b_*radius + doseMap_[layer].c_*std::pow(radius, 2)); //dose in rad
+  return cellDose/1000.; //convert to kRad
+}
+
+double HGCHEbackSignalScaler::getFluenceValue(int layer, float radius)
+{
+  double cellFluence = std::pow(10, doseMap_[layer].d_ + doseMap_[layer].e_*radius + doseMap_[layer].f_*std::pow(radius, 2)); //dose in rad
```

Comment:
another repeated computation is `std::pow(radius,2)` - you could do this once in the code below and pass in `radius2` to both functions